idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -331,214 +319,103 @@ def run_chunk_wrapper(chunks_in_flight, chunk_output_files, n, mutex, target, ar
-                            os.path.join(self.chunks_result_dir_s3, "")
-                        ]
-                    )
-                )
-            log.write(f"finished alignment for chunk {chunk_id} on {service} server {instance_ip}")
+            log.write(f"finished alignment for chunk {chunk_id} with {self.alignment_algorithm} by lazily fetching last result")
+            return multihit_local_outfile
+
+        # TODO: (tmorse) remove compat hack https://jira.czi.team/browse/IDSEQ-2568
+        deployment_environment = os.environ.get("DEPLOYMENT_ENVIRONMENT", self.additional_attributes.get("environment"))
+        priority_name = os.environ.get("PRIORITY_NAME", "normal")
+        provisioning_model = os.environ.get("PROVISIONING_MODEL", "EC2")
+
+        index_dir_suffix = self.additional_attributes["index_dir_suffix"]
+
+        job_name = f"idseq-{deployment_environment}-{self.alignment_algorithm}"
+        job_queue = f"idseq-{deployment_environment}-{self.alignment_algorithm}-{provisioning_model}-{index_dir_suffix}-{priority_name}"
+        job_definition = f"idseq-{deployment_environment}-{self.alignment_algorithm}"
+
+        environment = [{
+            'name': f"INPUT_PATH_{i}",
+            'value': os.path.join(self.chunks_result_dir_s3, input_file),
+        } for i, input_file in enumerate(input_files)] + [{
+            'name': "OUTPUT_PATH",
+            'value': multihit_s3_outfile,
+        }]
+
+        session = boto3.session.Session()
+        client = session.client("batch")
+        response = client.submit_job(
+            jobName=job_name,
+            jobQueue=job_queue,
+            jobDefinition=job_definition,
+            containerOverrides={"environment": environment},
+            retryStrategy={"attempts": CHUNK_MAX_ATTEMPTS},
+            timeout={"attemptDurationSeconds": CHUNK_ATTEMPT_TIMEOUT}
+        )
+        job_id = response["jobId"]
+
+        log.write(f"waiting for {self.alignment_algorithm} batch queue for chunk {chunk_id}.")
+        total_timeout = CHUNK_MAX_ATTEMPTS * CHUNK_ATTEMPT_TIMEOUT
+        end_time = time.time() + total_timeout
+        chunks_in_flight = min(chunk_count, MAX_CHUNKS_IN_FLIGHT)  # use min in case we have fewer chunks than the maximum
+        mean_delay = chunks_in_flight  # ~1 chunk per second to avoid throttling,
+        delay = mean_delay + random.randint(-mean_delay // 2, mean_delay // 2)  # Add some noise to de-synchronize chunks
+        status = "SUBMITTED"
+        while time.time() < end_time:
+            try:
+                status = self._get_job_status(session, job_id)
+            except ClientError as e:
+                # If we get throttled, randomly wait to de-synchronize the requests
+                if e.response['Error']['Code'] == "TooManyRequestsException":
+                    log.log_event("describe_jobs_rate_limit_error", values={"job_id": job_id}, warning=True)
+                    delay = delay ** 2  # exponential backoff
```

Comment:
@kislyuk you recommended removing this due to increased complexity. @tfrcarvalho mentioned it would be nice to have just in case. It is only one line so I think the difference is pretty marginal either way. I would slightly lean towards keeping in case of very high load situations but I don't feel that strongly. Any further thoughts based on the code here?

Reply:
I'm fine either way - leaving it up to you.